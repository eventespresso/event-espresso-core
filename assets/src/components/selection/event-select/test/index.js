import { shallow, render } from 'enzyme';
import { EventSelect } from '../index';

const simulatedEventOptions = [
	{ EVT_ID: 1, EVT_name: 'Event A' },
	{ EVT_ID: 2, EVT_name: 'Event B' },
];

describe( 'EventSelect with default options', () => {
	it( 'should render and match snapshot', () => {
		const selector = shallow( <EventSelect /> );
		expect( selector ).toMatchSnapshot();
	} );
} );

describe( 'EventSelect with no events and finished loading', () => {
	it( 'should render and match snapshot', () => {
		const selector = shallow( <EventSelect isLoading={ false } /> );
		expect( selector ).toMatchSnapshot();
	} );
} );

describe( 'EventSelect with 2 events', () => {
	const element = <EventSelect
		isLoading={ false }
		events={ simulatedEventOptions }
		onEventSelect={ jest.fn() }
	/>;
	it( 'should render and match snapshot', () => {
		const selector = shallow( element );
		expect( selector ).toMatchSnapshot();
	} );
	it( 'should render and have 2 options', () => {
		const selector = render( element );
		expect( selector.find( 'option' ) ).toHaveLength( 2 );
	} );

	it( 'should render and the first option has label and value matching the first siumlated event',
		() => {
			const selector = render( element );
			const firstOption = selector.find( 'option' ).first();
			expect( firstOption.text() )
				.toEqual( simulatedEventOptions[ 0 ].EVT_name );
			expect( firstOption.val() )
				.toEqual( simulatedEventOptions[ 0 ].EVT_ID.toString() );
		}
	);
} );
