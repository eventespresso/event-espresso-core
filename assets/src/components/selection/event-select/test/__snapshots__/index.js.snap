// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EventSelect with 2 events should render and match snapshot 1`] = `
<React.Fragment>
  <_class
    label="Select Event"
    onChange={[Function]}
    options={
      Array [
        Object {
          "label": "Event A",
          "value": 1,
        },
        Object {
          "label": "Event B",
          "value": 2,
        },
      ]
    }
    value={0}
  />
</React.Fragment>
`;

exports[`EventSelect with default options should render and match snapshot 1`] = `
<React.Fragment>
  <Placeholder
    icon="calendar"
    label="EventSelect"
  >
    <Component />
  </Placeholder>
</React.Fragment>
`;

exports[`EventSelect with no events and finished loading should render and match snapshot 1`] = `
<React.Fragment>
  <Placeholder
    icon="calendar"
    label="EventSelect"
  >
    There are no events to select from. You need to create an event first.
  </Placeholder>
</React.Fragment>
`;
